mysql/resource_grant.go

```diff
@@ -690,8 +716,29 @@ func setDataFromGrant(grant MySQLGrant, d *schema.ResourceData) *schema.Resource
 	return d
 }
 
-func hasConflictingGrants(ctx context.Context, db *sql.DB, desiredGrant MySQLGrant) (MySQLGrant, error) {
+func combineGrants(grantA MySQLGrant, grantB MySQLGrant) (MySQLGrant, error) {
+	// We can combine grants with privileges
+	grantAWithPrivileges, aOk := grantA.(MySQLGrantWithPrivileges)
```

Comment:
Consider comparing databases/tables for those privileges. I understand this will be correct with whole code flow, but I'd like to be able to trust function without looking at callers.